cmd/prerun.go

```diff
@@ -0,0 +1,95 @@
+package cmd
+
+import (
+	"io"
+	"os"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/astronomer/astro-cli/deployment"
+	"github.com/astronomer/astro-cli/houston"
+	"github.com/astronomer/astro-cli/messages"
+	"github.com/astronomer/astro-cli/version"
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+)
+
+// PersistentPreRunCheck for validation of the CLI vs Deployment Version
+func PersistentPreRunCheck(client *houston.Client, cmd *cobra.Command, out io.Writer) {
+	ac := deployment.GetAppConfig()
+
+	// Skip check if AppConfig is nil
+	if ac != nil {
+		dv := ac.Version
+		cv := version.CurrVersion
+
+		validateVersions(cv, dv)
+	}
+}
+
+func validateVersions(cv string, dv string) {
+	if isBehindMajor(cv, dv) {
+		color.Red(messages.ERROR_NEW_MAJOR_VERSION, cv, dv)
+		// Exit for commands that require matching major versions
+		os.Exit(1)
+	} else if isBehindPatch(cv, dv) {
+		color.Yellow(messages.WARNING_NEW_PATCH_VERSION, cv, dv)
+	} else if isAheadMajor(cv, dv) {
+		color.Yellow(messages.WARNING_DOWNGRADE_VERSION, cv, dv)
+	}
+}
+
+func isBehindMajor(cv string, dv string) bool {
+	fm := formatMajor(dv)
+	fc := formatLtConstraint(fm)
+	m := getConstraint(fc)
+	v := getVersion(cv)
```

Comment:
general note, try to avoid `m`, `v` etc. better short but concise

Reply:
I can change to `majCon` and `ver`?